assignment-3.md

```diff
@@ -0,0 +1,108 @@
+feature. Change this:
+
+```typescript
+test('Climate app shows the current temperature…', …)
+```
+
+to this:
+
+```typescript
+test.skip('Climate app shows the current temperature…', …)
+```
+
+## 3.2 Header
+
+Let's get started by writing some tests for the `Header` component that we
+have implemented in the first assignment. You have probably put that into a
+file called `Header.tsx`. Create a new file `Header.test.tsx` for the tests.
+
+Try to `render` the component with different text values that you pass in as a
+prop. Use `getByRole` and `toHaveTextContent` to find the rendered header in the
+DOM and check its text content.
+
+*Optional:* If you're feeling fancy, you can use
+[Jest's `test.each`](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout)
+to create a parametrized test.
+
+### Hints
+
+The [Testing Playground Chrome extension](https://chrome.google.com/webstore/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano)
+can help you with writing queries.
+
+The existing test in `Climate.test.tsx` will give you an example how a typical
+test looks like. Don't copy over too much, though! The `Climate` component has
+to deal with some asynchrony, as it is listening to sensor events.
+
+Avoid using any of the following in `Header.test.tsx`:
+
+* `sensor` / `testSensor`
+* `async` / `await`
+* `findBy…` queries
+
+## 3.3 Current Temperature and Humidity
+
+Back to `Climate.test.tsx`. Remove that `skip` you added in 3.1 and get the test
+working again!
+
+When it works, add a similar test for the current humidity value.
+
+### Hints
+
+You have introduced a child component in assignment 3.2 and are using that in
+your `Climate` component. We are still just writing tests on the level of the
+`Climate` component and not on the child component though. That is not a unit
+test, but an integration test, and it's the approach we want to use here.
+
+If you have trouble writing a good query to find the current temperature and
+humidity values, change your component so that it adds `data-testid` attributes.
+Make sure to assign unique values to them. You can then use `findByTestId` in
+your tests.
+
+## 3.4 Minimum and Maximum Values
+
+Add tests for minimum and maximum values. It is sufficient to do this for either
+the temperature or the humidity here.
+
+### Hints
+
+Let the `testSensor` emit appropriate values so that you can actually be sure
+that the minimum and maximum only change when the current value exceeds them.
+
+## 3.5 Reset Button
+
+Test that the reset button is doing what it should. Also make sure that after
+the button is pressed the minimum and maximum are being updated again when new
+values are emitted by the sensor.
+
+Try to avoid using a `data-testid` this time. Query for the button with
+`findByRole` and a `name` option. The *Testing Playground* extension can help
+you again. If both of your buttons have the same text, change that! It makes
```

Comment:
the test implies that there should be two reset buttons, while the assignment 2 does not specifically mention this.
(it even explicitly states "add a reset button")

Reply:
Good catch!

Reply:
I've just removed it.